test/plugin/base_test.rb

```diff
@@ -704,7 +704,7 @@ def test_split_logs_by_tag
     end
   end
 
-  def test_timestamps
+  def test_adjusted_timestamps
```

Comment:
The tests `test_adjusted_timestamps` and `test_unadjusted_timestamps` look quite similar except the expected timestamp and the timestamp configuration.

How about something like

```
    [
      [true, Time.at(123_456.789), Time.at(123_456.789)],
      [true, Time.at(0), Time.at(0)],
      [true, current_time, current_time],
      [true, one_second_before_next_year, adjusted_to_last_year],
      [true, next_year, Time.at(0)],
      [true, one_second_into_next_year, Time.at(0)],
      [true, one_day_into_next_yearn Time.at(0)],
      [false, Time.at(123_456.789), Time.at(123_456.789)],
      [false, Time.at(0), Time.at(0)],
      [false, current_time, current_time],
      [false, one_second_before_next_year, one_second_before_next_year],
      [false, next_year, next_year],
      [false, one_second_into_next_year, one_second_into_next_year],
      [false, one_day_into_next_year, one_day_into_next_year]
    ].each do |adjust_ts_config, original_ts, expected_ts|
```

?

Reply:
To be more accurate, I meant
```
    [
      [APPLICATION_DEFAULT_CONFIG, Time.at(123_456.789), Time.at(123_456.789)],
      [APPLICATION_DEFAULT_CONFIG, Time.at(0), Time.at(0)],
      [APPLICATION_DEFAULT_CONFIG, current_time, current_time],
      [APPLICATION_DEFAULT_CONFIG, one_second_before_next_year,
       adjusted_to_last_year],
      [APPLICATION_DEFAULT_CONFIG, next_year, Time.at(0)],
      [APPLICATION_DEFAULT_CONFIG, one_second_into_next_year, Time.at(0)],
      [APPLICATION_DEFAULT_CONFIG, one_day_into_next_yearn Time.at(0)],
      [NO_ADJUST_TIMESTAMPS_CONFIG, Time.at(123_456.789), Time.at(123_456.789)],
      [NO_ADJUST_TIMESTAMPS_CONFIG, Time.at(0), Time.at(0)],
      [NO_ADJUST_TIMESTAMPS_CONFIG, current_time, current_time],
      [NO_ADJUST_TIMESTAMPS_CONFIG, one_second_before_next_year,
       one_second_before_next_year],
      [NO_ADJUST_TIMESTAMPS_CONFIG, next_year, next_year],
      [NO_ADJUST_TIMESTAMPS_CONFIG, one_second_into_next_year,
       one_second_into_next_year],
      [NO_ADJUST_TIMESTAMPS_CONFIG, one_day_into_next_year,
       one_day_into_next_year]
    ].each do |(adjust_ts_config, ts, expected_ts)|
```

Reply:
Did something similar. PTAL.